lib/packwerk/parse_run.rb

```diff
@@ -45,6 +46,12 @@ def detect_stale_violations
     sig { returns(Result) }
     def update_deprecations
       offense_collection = find_offenses
+
+      @run_context.package_set.each do |package|
```

Comment:
I wonder if we could do a small tidy to reuse some things in `lib/packwerk/offense_collection.rb`...

I was thinking we could have a method:
```
# replaces `deprecated_references_for`, a private method in `OffenseCollection`
deprecated_references = Packwerk::DeprecatedReferences.for(package)
```

Then we could call:
```
deprecated_references.delete_if_exists
```

Which is of course just:
```
File.delete(@filepath) if File.exist?(@filepath)
```

Also note that in `DeprecatedRefererences#dump`, I think we can remove the `if @new_entries.empty` portion given this new implementation.

Reply:
I'd really like to be able to just be able to do something like `package_set.each { |package| package.deprecated_references.persist! }` but I think we're just too far away from that. I can do the change you suggested, it does at least centralize the deprecated reference file IO in the one class which I like. Thanks for taking the time to look at this!